test/metabase/test_runner/junit/write.clj

```diff
@@ -42,7 +42,7 @@
 (defn- print-result-description [{:keys [file line message testing-contexts], :as result}]
   (println (format "%s:%d" file line))
   (doseq [s (reverse testing-contexts)]
-    (println (str/trim (decolorize-and-escape s))))
+    (println (str/trim (decolorize-and-escape (str s)))))
```

Comment:
Not 100% sure we need this but the new `with-native-query-testing-context` stuff doesn't actually calculate the pretty-formatted info unless you call `(.toString)` on it. This is so we don't waste a bunch of time for tests that pass -- only does the expensive pretty stuff when it prints `testing` context (e.g. on test failures)